packages/netsuite-adapter/src/client/client.ts

```diff
@@ -203,6 +205,47 @@ export default class NetsuiteClient {
       .toArray()
   }
 
+  private static createDependencyMap(
+    changesData: ReadonlyArray<ChangeDataType>
+  ):Record<string, string[]> {
+    const dependencyMap: Record<string, string[]> = {}
+    changesData
+      .filter(isInstanceElement)
+      .forEach(instance => walkOnElement({
+        element: instance,
+        func: ({ value }) => {
+          if (scriptidReferenceIdRegex.test(value)) {
+            const currentScriptId = value.split('=')[1].slice(0, -1)
+            const currentElement = instance.elemID.getFullName()
+            if (!(currentElement in dependencyMap)) {
+              dependencyMap[currentElement] = [currentScriptId]
+            } else {
+              dependencyMap[currentElement].push(currentScriptId)
+            }
+          }
+          return WALK_NEXT_STEP.RECURSE
+        },
+      }))
+    return dependencyMap
+  }
+
+  private static getManifestErrorElemIdName(
+    error: ManifestValidationError,
+    dependencyMap: Record<string, string[]>,
+    changes: ReadonlyArray<Change>
+  ): Set<string> {
+    const changeData = changes.map(getChangeData)
+    const elementsToRemoveElemIDs = new Set<string>()
+    Object.keys(dependencyMap).forEach(key => {
+      if (error.errorScriptIds?.some(scriptid => dependencyMap[key].includes(scriptid))) {
+        elementsToRemoveElemIDs.add(key)
+      }
+    })
+    return elementsToRemoveElemIDs.size === 0
+      ? new Set(changeData.map(elem => elem.elemID.getFullName()))
+      : elementsToRemoveElemIDs
```

Comment:
Shouldn't it be just like the below?
I don't understand the logic here.

```suggestion
    return elementsToRemoveElemIDs
```

Reply:
This covers the case where we cannot find the cause of the error and do not remove any changes from `changesToDeploy` so we should remove all changes and fail the deploy, otherwise we are stuck re-deploying/validating and nothing changes. @omrilit